### PR TITLE
[TS] LPS-151455 Set buffered image type to TYPE_INT_ARGB if its type is TYPE_BYTE_INDEXED

### PR DESCRIPTION
**Pull request by evibudai**
Hi Team,

[LPS-151455](https://issues.liferay.com/browse/LPS-151455)

**Issue:**
When scaling an image to preview size, using java.awt.image.BufferdImage for creating the new image, in case of the image has 8bit color model (the type is TYPE_BYTE_INDEXED) based on the implementation limitation the new color model is losing the alpha and transparency properties. So the originally transparent background turns to black.

**Solution:**
I created a fix based on this PTR: [PTR-2983](https://issues.liferay.com/browse/PTR-2983)
in com.liferay.portal.image.ImageToolImpl#doScale doing this:
`if (type == BufferedImage.TYPE_CUSTOM) { type = BufferedImage.TYPE_INT_ARGB; }`
if we change the condition to 
`if (type == BufferedImage.TYPE_CUSTOM ||
 type == BufferedImage.TYPE_BYTE_INDEXED) {
type = BufferedImage.TYPE_INT_ARGB;
}`

Now the preview of the uploaded image stays transparent. I tested it, and it doesn't have side effects.

Could you please review it?
Thank you!

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/image/ImageToolImpl.java [hunk 1]
```diff
@@ -734,7 +734,9 @@ protected RenderedImage doScale(
 
 		int type = originalBufferedImage.getType();
 
-		if (type == BufferedImage.TYPE_CUSTOM) {
+		if ((type == BufferedImage.TYPE_CUSTOM) ||
```
Comment (AliciaGarciaGarcia) [inline portal-impl/src/com/liferay/portal/image/ImageToolImpl.java:737]: the if with the check of the  : _BufferedImage.TYPE_BYTE_INDEXED_ and _BufferedImage.TYPE_CUSTOM_ maybe should ordered by letter.

if the order is significant maybe must tell Brian about it or separate on 2 ifs

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portal/image/ImageToolImpl.java (hunk 1 above)
Comment (evibudai) [reply portal-impl/src/com/liferay/portal/image/ImageToolImpl.java]: Hi @AliciaGarciaGarcia,
the type has one integer value, so it matches one bufferedImage type or neither, so the order is not significant.

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**